modules/tinymce/src/core/main/ts/content/SetContentImpl.ts

```diff
@@ -19,9 +19,28 @@ import * as NodeType from '../dom/NodeType';
 import * as EditorFocus from '../focus/EditorFocus';
 import * as FilterNode from '../html/FilterNode';
 import { Content, SetContentArgs } from './ContentTypes';
+import { SetContentFormatter } from './EditorContent';
 
 const defaultFormat = 'html';
 
+const defaultContentFormatter = (editor: Editor, content: Content, updatedArgs: SetContentArgs) => {
+  if (!isTreeNode(content)) {
+    content = updatedArgs.content;
+  }
+
+  return Optional.from(editor.getBody()).fold(
+    Fun.constant(content),
+    (body) => isTreeNode(content) ? setContentTree(editor, body, content, updatedArgs) : setContentString(editor, body, content, updatedArgs)
+  );
+};
+
+const addDefaultSetFormats = (formatCell: Cell<Record<string, SetContentFormatter>>) => {
+  addSetContentFormatter('raw', defaultContentFormatter, formatCell);
+  addSetContentFormatter('text', defaultContentFormatter, formatCell);
+  addSetContentFormatter('html', defaultContentFormatter, formatCell);
+  addSetContentFormatter('tree', defaultContentFormatter, formatCell);
```

Comment:
See my previous comments on the getter about duplication and making `tree` a separate formatter since it deals with `AstNode` instead of `string`.